packages/server/src/server/handlers/observability.test.ts

```diff
@@ -0,0 +1,336 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import type { Mastra } from '@mastra/core/mastra';
+import type { MastraStorage } from '@mastra/core/storage';
+import type { AITrace } from '@mastra/core/storage';
+import { AISpanType } from '@mastra/core/ai-tracing';
+import { HTTPException } from '../http-exception';
+import { getAITraceHandler, getAITracesPaginatedHandler } from './observability';
+
+// Utility function to create test spans with sensible defaults
+const createTestSpan = (overrides: Partial<any> = {}) => ({
+  id: 'test-trace-test-span',
```

Comment:
lets remove setting `id` in the creator, and instead just create it from a join of `traceId` and `spanId`... makes test updating easier.